crates/ruff_linter/src/rules/flake8_pyi/rules/redundant_literal_union.rs

```diff
@@ -145,17 +147,22 @@ fn match_builtin_type(expr: &Expr, semantic: &SemanticModel) -> Option<ExprType>
 
 /// Return the [`ExprType`] of an [`Expr]` if it is a constant (e.g., an `int`, like `1`, or a
 /// `bool`, like `True`).
-fn match_constant_type(expr: &Expr) -> Option<ExprType> {
-    let result = match expr {
-        Expr::BooleanLiteral(_) => ExprType::Bool,
-        Expr::StringLiteral(_) => ExprType::Str,
-        Expr::BytesLiteral(_) => ExprType::Bytes,
-        Expr::NumberLiteral(ast::ExprNumberLiteral { value, .. }) => match value {
+fn match_literal_type(expr: &Expr) -> Option<ExprType> {
+    let Some(literal_expr) = expr.as_literal_expr() else {
+        return None;
+    };
+    let result = match literal_expr {
+        LiteralExpressionRef::BooleanLiteral(_) => ExprType::Bool,
+        LiteralExpressionRef::StringLiteral(_) => ExprType::Str,
+        LiteralExpressionRef::BytesLiteral(_) => ExprType::Bytes,
+        LiteralExpressionRef::NumberLiteral(ast::ExprNumberLiteral { value, .. }) => match value {
             ast::Number::Int(_) => ExprType::Int,
             ast::Number::Float(_) => ExprType::Float,
             ast::Number::Complex { .. } => ExprType::Complex,
         },
-        _ => return None,
+        LiteralExpressionRef::NoneLiteral(_) | LiteralExpressionRef::EllipsisLiteral(_) => {
+            return None;
+        }
```

Comment:
I've explicitly mentioned to last 2 variants even though it's not strictly required.